src/rust-crypto/backup.ts

```diff
@@ -490,10 +493,15 @@ export class RustBackupDecryptor implements BackupDecryptor {
 export type RustBackupCryptoEvents =
     | CryptoEvent.KeyBackupStatus
     | CryptoEvent.KeyBackupSessionsRemaining
-    | CryptoEvent.KeyBackupFailed;
+    | CryptoEvent.KeyBackupFailed
+    | CryptoEvent.KeyBackupPrivateKeyCached;
 
 export type RustBackupCryptoEventMap = {
     [CryptoEvent.KeyBackupStatus]: (enabled: boolean) => void;
     [CryptoEvent.KeyBackupSessionsRemaining]: (remaining: number) => void;
     [CryptoEvent.KeyBackupFailed]: (errCode: string) => void;
+    /**
+     * Fired when the backup decryption key is received via secret sharing and stored in cache.
```

Comment:
aren't there other ways a decryption key can be received?

Also, we probably don't need to document again what the event means: it is sufficient to put the documentation on `CryptoEvent.KeyBackupPrivateKeyCached` IMHO.

Reply:
yes, updated.
BTW, the current system where we re-emit things is a bit hard to maintain, we have to declare the events at several places. Not sure how to do better.